api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,133 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	deployID := generateEntityID(req.DeployName)
+	fqDeployID := addLayer0Prefix(d.Config.Instance(), deployID)
+	familyName := fqDeployID
+
+	taskDefinition, err := d.renderTaskDefinition(req.DeployFile, familyName)
+	if err != nil {
+		return nil, err
+	}
+
+	taskDefinitionOutput, err := d.createTaskDefinition(taskDefinition)
```

Comment:
[nit] this returns a `taskDefinition`, not a `taskDefinitionOutput` (since that object only exists in the helper)

Reply:
gonna skip that one as we have a `taskDefinition` from `renderTaskDefinition` already. `Output` maybe isn't the greatest suffix in the world, but it helps delineate the return from `createTaskDefinition`.